sys/cddl/contrib/opensolaris/uts/common/dtrace/fasttrap.c

```diff
@@ -2248,7 +2248,7 @@ fasttrap_ioctl(struct cdev *dev, u_long cmd, caddr_t arg, int fflag,
 		size_t size;
 		int ret, err;
 
-#ifdef COMPAT_CHERIABI
+#if __has_feature(capabilities)
 		if (SV_PROC_FLAG(td->td_proc, SV_CHERI))
 			uprobe = *(fasttrap_probe_spec_t * __capability *)arg;
 		else
```

Comment:
In terms of reducing the diff to upstream, I would almost be tempted to invert the test and `__USER_CAP` the odd case.  I would also stick it under `#ifdef COMPAT_FREEBSD64` since you have to make an assumption about what `arg` points to, and I would make it assume a `uint64_t` to be purecap-kernel friendly:

```
#ifdef COMPAT_FREEBSD64
    if (!SV_PROC_FLAG(td->td_proc, SV_CHERI))
        uprobe = __USER_CAP(*(uint64_t *)arg, sizeof(...));
    else
#endif
        uprobe = *(fasttrap_probe_spec_t * __capability *)arg;
```

I think that approach might also be true of some of the other changes in this PR if you agree?